php/libraries/NDB_Menu_Filter.class.inc

```diff
@@ -145,7 +145,10 @@ class NDB_Menu_Filter extends NDB_Menu
         $success = $this->_setupVariables();
 
         // set the headers if necessary
-        if (!is_array($this->headers) || count($this->headers) == 0) {
+        if ((!is_array($this->headers)
+            || count($this->headers) == 0)
+            && is_array($this->columns)
```

Comment:
```suggestion
            && is_iterable($this->columns)
```

sorry, I should have caught this in the last review.. the requirement is that it's a type which is iterable using foreach, not that it's an array.

Reply:
done